### PR TITLE
Docker build

### PR DESCRIPTION
**Pull request by SmithRWORNL**
Adds a Github Action build system to produce a Docker image with subhkl installed inside it.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -3,9 +3,7 @@ name = "subhkl"
 description = "Solving crystal orientation from Laue images"
 dynamic = ["version"]
 requires-python = ">=3.10"
-dependencies = [
-    "mantidworkbench >= 6.7.20230830",
```
Comment (marshallmcdonnell) [inline pyproject.toml:7]: Quick check, this is not in PyPI that I am aware of [check search here](https://pypi.org/search/?q=mantidworkbench) an only available as a conda environment [see conda package here](https://anaconda.org/mantid/mantidworkbench).

So it makes sense to remove this from the pyproject toml.

Just sort of justifying removing the dependency here and instead using the conda and / or docker installed mantid workbench version :+1:

File: .github/workflows/github-actions.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    steps:
+      - name: Download repo
+        uses: actions/checkout@v4
+
+      - name: Code ornl gov login
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: code.ornl.gov:4567
+          password: ${{ secrets.rse_mantid_docker_token }}
+          username: ${{ secrets.rse_mantid_docker_username }}
```
Comment (marshallmcdonnell) [inline .github/workflows/github-actions.yaml:28]: Just to check, is this username and token tied to your credentials @SmithRWORNL?

Or a project based token?

**Review by marshallmcdonnell — APPROVED** [review]
I have just one question but overall, LGTM!

Thank you, Robert!

File: .github/workflows/github-actions.yaml (hunk 2 above)
Comment (SmithRWORNL) [reply .github/workflows/github-actions.yaml:28]: @marshallmcdonnell My username, with a project based token.

Comment (marshallmcdonnell) [reply .github/workflows/github-actions.yaml:28]: Awesome :+1: 

We also have a service account we can swap to for other projects.

I'll send it via a direct communication.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ADD . /build
+
+RUN pdm build
+
+FROM code.ornl.gov:4567/rse/images/mantid-framework:6.8.20231027.1822-py3.10 as tool
+
+COPY --from=build /build/dist/subhkl-0.1.0-py3-none-any.whl subhkl-0.1.0-py3-none-any.whl
+RUN pip install subhkl-0.1.0-py3-none-any.whl
+
+CMD [ "bash" ]
```
[the remark this answers is not in the document]
Comment (SmithRWORNL) [reply Dockerfile:25]: 1) We can, it looks like test_optimization.py has some asserts in it, so we can include an automated run of it.
2) Currently, we don't have a CLI. We need some agreement of what process to use once the .h5 file is loaded and how to then take those results and make a .mtz file from them in order to create container that can actually be run.
3) Will do

[the remark this answers is not in the document]
Comment (SmithRWORNL) [reply Dockerfile:29]: @marshallmcdonnell Latest commit addresses 1 and 3. We'll need to discuss what exactly needs to be done with the code to produce the data we want to tackle 2.

**Comment by marshallmcdonnell** [comment]
I'm happy if we merge this in for now.
Thank you @SmithRWORNL 

@zjmorgan we probably do need to know the CLI interface we need to create for the `.h5` -> `.mtz` tooling.

Yet, we can do that in another change set.

**Comment by marshallmcdonnell** [comment]
@zjmorgan Just checking, is it okay to merge this in?

**Comment by zjmorgan** [comment]
I think I generate it on the fly from the image tif data

**Comment by marshallmcdonnell** [comment]
@SmithRWORNL Thank you! Great work!

Just to check, is this ready for merge?
If so, I can sync w/ @zjmorgan as needed 

Thanks for creating this issue related to this work: https://github.com/zjmorgan/subhkl/issues/6

**Comment by SmithRWORNL** [comment]
@marshallmcdonnell Yes, it can be merged now.

**Pull request merged**